scripts/ci/ws12_workflow_contracts.py

```
@@ -183,6 +183,123 @@ def validate_e2e_scope_filters(text: str) -> list[str]:
+            f"{CODE_STYLE_WORKFLOW}: could not find the {PRODUCTION_LINT_STEP!r} step "
+            "— it is the only clippy gate on pull requests and must stay assertable"
+        ]
+    # Comments in the step explain which flags are absent and why, so they name
+    # the very strings being rejected.
+    command = "\n".join(
+        line for line in body.splitlines() if not line.lstrip().startswith("#")
+    )
+    if "cargo clippy" not in command:
+        return [
+            f"{CODE_STYLE_WORKFLOW}: {PRODUCTION_LINT_STEP!r} no longer runs "
+            "`cargo clippy` — this contract can only pin a command it can see"
```

Ruff (0.16.1)

[warning] Unparenthesized implicit string concatenation in collection

Did you forget a comma?

(ISC004)